File: Old C# Codes/DebateMatch.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DebateTournamentTabSystem.BLL
{
    public class DebateMatch
    {
        public DebateTeam TeamA { get; set; }
        public DebateTeam TeamB { get; set; }
        public int[] TeamAScores { get; set; } = new int[3];
        public int[] TeamBScores { get; set; } = new int[3];
        public bool IsCompleted { get; set; } = false;

        public DebateMatch(DebateTeam a, DebateTeam b)
        {
            TeamA = a;
            TeamB = b;
        }

        public void SubmitScores(int[] aScores, int aRebuttal, int[] bScores, int bRebuttal)
        {
            if (aScores.Length != 3 || bScores.Length != 3)
                throw new ArgumentException("Each team must have exactly 3 debater scores.");

            TeamAScores = aScores;
            TeamBScores = bScores;

            int teamATotal = aScores.Sum() + aRebuttal;
            int teamBTotal = bScores.Sum() + bRebuttal;

            for (int i = 0; i < 3; i++)
            {
                TeamA.teamMembers[i].increaseIndividualScore(aScores[i]);
                TeamB.teamMembers[i].increaseIndividualScore(bScores[i]);
            }

            TeamA.increaseTeamScore(teamATotal);
            TeamB.increaseTeamScore(teamBTotal);

            if (teamATotal > teamBTotal)
            {
                TeamA.teamWinsADebate();
                TeamB.teamLosesADebate();
            }
            else if (teamBTotal > teamATotal)
            {
                TeamB.teamWinsADebate();
                TeamA.teamLosesADebate();
            }
            else
            {
                Console.WriteLine("The scores are tied.");
                Console.WriteLine($"Team A: {TeamA.teamName}");
                Console.WriteLine($"Team B: {TeamB.teamName}");
                Console.WriteLine("Enter the number of the team that wins (1 for Team A, 2 for Team B):");

                while (true)
                {
                    string? input = Console.ReadLine();
                    if (input == "1")
                    {
                        TeamA.teamWinsADebate();
                        TeamB.teamLosesADebate();
                        break;
                    }
                    else if (input == "2")
                    {
                        TeamB.teamWinsADebate();
                        TeamA.teamLosesADebate();
                        break;
                    }
                    else
                    {
                        Console.WriteLine("Invalid input. Please enter 1 or 2:");
                    }
                }
            }

            IsCompleted = true;
        }
    }
}

File: lib/models/debate_match.dart
import 'debate_team.dart';

class DebateMatch {
  DebateTeam teamA, teamB;
  List<int> teamAScores, teamBScores;
  bool isCompleted;
  int? venue;

  DebateMatch({
    required this.teamA,
    required this.teamB,
    this.teamAScores = const [0, 0, 0],
    this.teamBScores = const [0, 0, 0],
    this.isCompleted = false,
    this.venue,
  });

  void submitScores(List<int> scoresA, List<int> scoresB, int rebA, int rebB) {
    for (int i = 0; i < 3; i++) {
      teamA.teamMembers[i].increaseDebaterScore(scoresA[i]);
      teamB.teamMembers[i].increaseDebaterScore(scoresB[i]);
    }
    int totalA = scoresA[0] + scoresA[1] + scoresA[2] + rebA;
    int totalB = scoresB[0] + scoresB[1] + scoresB[2] + rebB;
    teamA.increaseTeamScore(totalA);
    teamB.increaseTeamScore(totalB);
    if (totalA > totalB) {
      teamA.teamWinsADebate();
      teamB.teamLosesADebate();
    } else if (totalB > totalA) {
      teamB.teamWinsADebate();
      teamA.teamLosesADebate();
    } else {
      throw Exception(
          "Check Tie: ${teamA.teamName} scores $totalA vs ${teamB.teamName} scores $totalB");
    }
    isCompleted = true;
  }

  // Convert to JSON for storage
  Map<String, dynamic> toJson() {
    return {
      'teamA': teamA.toJson(),
      'teamB': teamB.toJson(),
      'teamAScores': teamAScores,
      'teamBScores': teamBScores,
      'isCompleted': isCompleted,
      'venue': venue,
    };
  }

  // Create from JSON
  factory DebateMatch.fromJson(Map<String, dynamic> json) {
    return DebateMatch(
      teamA: DebateTeam.fromJson(json['teamA'] ?? {}),
      teamB: DebateTeam.fromJson(json['teamB'] ?? {}),
      teamAScores: List<int>.from(json['teamAScores'] ?? [0, 0, 0]),
      teamBScores: List<int>.from(json['teamBScores'] ?? [0, 0, 0]),
      isCompleted: json['isCompleted'] ?? false,
      venue: json['venue'],
    );
  }
}
